### DISCUSSION TITLE
Sometimes in games, the picture freezes completely and there is only sound and I can control the character

### DISCUSSION THREAD

**Discussion by LibertyGM** [LizardByte/community#608] [Sunshine]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Is your issue described in the documentation?

- [x] I have read the documentation

### Is your issue present in the latest beta/pre-release?

None

### Describe the Bug

I use sunshine and moonlight on Arch Linux. Sometimes in games, the picture freezes completely and there is only sound and I can control the character (the picture on the screen remains frozen). The problem occurs in several games. It has been found recently in PUBG and Escape From Tarkov. Especially often in Escape From Tarkov. I load into the raid and the image freezes, but there are sounds, and if I press the keys, I hear the character moving.

sunshine here:
Windows 11
Processor: i7-10700k
Graphics Processor: NVIDIA GeForce RTX 3070Ti

moonlight here:
Operating System: Arch Linux 
KDE Plasma Version: 6.2.4
KDE Frameworks Version: 6.9.0
Qt Version: 6.8.1
Kernel Version: 6.12.7-arch1-1 (64-bit)
Graphics Platform: Wayland
Processors: 32 × Intel® Core™ i9-14900KF
Memory: 62.6 ГиБ of RAM
Graphics Processor: NVIDIA GeForce RTX 4080/PCIe/SSE2
Manufacturer: ASUS

### Host Operating System

Linux

### Operating System Version

Arch Linux

### Architecture

amd64/x86_64

### Sunshine commit or version

Latest beta

### Package

Linux - AUR (Third Party)

### GPU Type

NVIDIA

### GPU Model

4080

### GPU Driver/Mesa Version

Latest

### Capture Method

None

### Config

```Shell

```

### Apps

```json

```

### Relevant log output

```shell
Everything freezes, I don't know how to get the log
```

**Comment by chrism212** [comment]
I have this same issue.  Windows Sunshine host with Macbook Moonlight client.  It seems to be much more likely to happen with specific games (Monster Hunter Wilds and Diablo 4 for me.)

**Comment by Befeeter** [comment]
same for me today. was not happened since today from never and i was using it every day, maybe mh wilds related.

EDIT: I have make some test and is related to the HAGS option in Windows 11. i think Monster Hunter Wilds is hard on VRAM demand and it's caussing the issue.

i have a 3080Ti with 12GB of VRAM, do you know some way to have a little extra free vram or any workaround?

BR.

**Reply by Befeeter** [reply]
Update: Finally was associated a VRAM limitations, for my the solution was reduce the texture quality in game (MH Wilds) from the ULTRA HD textures to just high. 

It's OK for me because when i play with moonlight/sunshine in the 11" tablet, i can't appreciate the difference between the Ultra HD textures and High. 

Thanks for your replys.

BR.

**Comment by goldenpipes** [comment]
I've had this happen with windows 11 also,  I think it has something to do with display sleep timing or something, i installed PowerToys and set a Awake to "indefinitely" and it no longer freezes for me.

**Comment by Eyji** [comment]
I experienced this issue while playing God of War on my Steam Deck (as the client) with a Windows 11 computer as the host. I received several slow connection warnings, and then the game completely froze.

**Comment by sainsley1** [comment]
I think this is actually a bug with Sunshine.  My sunshine service rebooted after the last time this happened and then I had to manually restart it before it would stream video again.  I upgraded to latest release recently and was not having this issue prior, and immediately after the upgrade, started having this problem.  And I am on Win10.  I am going to downgrade a couple builds and see if the issue goes away.  If there is anything you want to troubleshoot the issue just let me know

**Comment by BuckRogers25** [comment]
Hi all, exactly the same issue as the title of this post here to.  Couldn't find an issue log for this, so maybe time to start one.  This is just a question post right now.

New install on a machine that is windows 11 now with latest version of sunshine and nvidia drivers, was on 10 before with no issue.  Can see the post above from @sainsley1 though and @LibertyGM on Arch so sounds like it isn't OS dependant, my sunshine on win 10 was probably a late in 2024 version and I was on win 10 22h2 with nvidia drivers always up to date (3080ti).  How did you get on with your downgrade @sainsley1 ? I may give that a go now and aim to go back to a late 2024 version and work forwards.  Looks like sunshine, but could be an nvidia driver issue to, might have to try downgrading both or one or the other to figure out which it is.  Seem the common thread is the HAGG settings and nvidia cards.

I changed the HAGG setting "Use Realtime priority in hardware accelerated gpu scheduling" to disable it after reading through the config options looking for a fix, and that does stop the freezes, but then you get a pretty nasty stutter at times.  So does seem to be the same high usage issue others have described here, and that workaround isn't a pleasant one.  Most of my testing was in DiabloIV for reference.  

I had a similar set of attempts, restarting sunshine fixes it for a moment or two, then starts again.  Rebooting the machine seems to fix it for around 30 minutes, then again.  I have PowerToys Awake on my machine to @goldenpipes so will try your suggestion as another workaround for now, instead of disabling the setting I just mentioned.  

I also got an alert from some monitoring software I use call Pulseway, which prompted me to check windows logs, there is an application error thrown as soon as a freeze occurs.  Here it is:

Faulting application name: Sunshine.exe, version: 0.0.0.0, time stamp: 0x679101d7
Faulting module name: KERNELBASE.dll, version: 10.0.26100.3470, time stamp: 0xcdfc8fa1
Exception code: 0x80000003
Fault offset: 0x000000000011e702
Faulting process id: 0xb580
Faulting application start time: 0x1dba34520a0b130
Faulting application path: C:\Program Files\Sunshine\Sunshine.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report Id: 44244643-6e18-4ea6-80d2-f8d121a6eb70
Faulting package full name: 
Faulting package-relative application ID:.

From that I would assume it is sunshine faulting but can't be certain, I'm no debug expert or issue hunter.  I couldn't see any NVidia related errors after trawling event logs, something from NVContainer or similar might have pointed to a driver problem instead.  I've enabled debug logging in sunshine and was going to force another and then post the log and my own issue breakdown as per the guidelines as @LibertyGM has.

For reference @LibertyGM  to get the log, go into sunshine settings, go to the General section and set your log level to "Debug" and apply, then go to Config Files section and set Logfile Path to something like "C:\logs\sunshine.log".  The log is overwritten on restart of sunshine, so when you see the freeze happen, go to your machine, and then copy the logfile somewhere else to post later.  I went to do this the other evening and then being tired restarted sunshine without even thinking, so now I shall have to do it again :).

Going to go off and try and reproduce this now and hope to post later.  After which will try @goldenpipes workaround.
Cheers.

**Reply by BuckRogers25** [reply]
Side note, just noticed you can view the log live also @ https://localhost:47990/troubleshooting .  

Got through my first test last night which went as expected and bombed out at roughly 30 minutes, and then fell asleep part way through test 2, so will complete my second test with @goldenpipes awake suggestion set to 60 minutes now and see if that changes things at all.

Timestamps in logs showing session start @ 21:35:22 and ended with "no pending sessions" @ 22:02:09.  I wont attach logs for now, just snippets, and post up full information later after I've completed the full test series and rest of the info gathering.

The end of the log from that test was:
> [2025-04-05 22:02:02.304]: Debug: NvEnc: encoded frame sizes in kB (min/max/avg): 21.60/96.41/88.36
> [2025-04-05 22:02:04.272]: Debug: Network: each send_batch() latency (min/max/avg): 0.04ms/13.13ms/0.52ms
> [2025-04-05 22:02:04.388]: Debug: Network: each FEC block latency (min/max/avg): 0.03ms/0.19ms/0.04ms
> [2025-04-05 22:02:04.389]: Debug: Network: frame's overall network latency (min/max/avg): 0.75ms/13.45ms/1.15ms
> [2025-04-05 22:02:09.982]: Debug: Frame capture sleep overshoot (min/max/avg): 0.02ms/0.78ms/0.28ms
> [2025-04-05 22:02:09.990]: Debug: Frame processing latency (min/max/avg): 6.30ms/129.30ms/7.70ms
> [2025-04-05 22:02:16.421]: Debug: No pending session for incoming RTSP connection
> [2025-04-05 22:02:47.437]: Debug: No pending session for incoming RTSP connection
> [2025-04-05 22:03:18.453]: Debug: No pending session for incoming RTSP connection

So it seemed to bomb with the client trying to reconnect with the same session, but no information in the log about the session being torn down on the host side, or keep alive notes showing the session reconnecting.

I did find an NVLDDMKM error in logs at the same time as the Application Error for sunshine, but I also messed up and restarted sunshine before beginning my second test which left the session open from the moonlight client reconnecting to the host and could account for that event log error.  Can see in the logs the client did reconnect automatically on service restart, but was 30 seconds after resuming the session in moonlight before the issue appeared again, and the logs didn't show any frame information for that session resuming either.  Will restart the test clean today and compare.

Here is a snippet from that not quite done right test after service restart, showing the client reconnecting after the service start section of the log is complete, but not frame info, just the keep alive.  Couldn't see anything out of place in that first part of the log. Have redacted my local ip below:

> [2025-04-05 22:31:40.725]: Debug: /CN=NVIDIA GameStream Client -- verified
> [2025-04-05 22:31:40.726]: Debug: TUNNEL :: HTTPS
> [2025-04-05 22:31:40.726]: Debug: METHOD :: GET
> [2025-04-05 22:31:40.726]: Debug: DESTINATION :: /serverinfo
> [2025-04-05 22:31:40.726]: Debug: User-Agent -- okhttp/3.12.13
> [2025-04-05 22:31:40.726]: Debug: Accept-Encoding -- gzip
> [2025-04-05 22:31:40.726]: Debug: Connection -- Keep-Alive
> [2025-04-05 22:31:40.726]: Debug: Host -- xxx.xxx.xxx.xxx:47984
> [2025-04-05 22:31:40.726]: Debug:  [--] 
> [2025-04-05 22:31:40.726]: Debug: uuid -- 27e6df3d-e2d6-4454-b889-0dd4681a7aac
> [2025-04-05 22:31:40.726]: Debug: uniqueid -- 0123456789ABCDEF

And here is from further on in the same log, after moonlight client reconnect, when the issue recurs.  You can see the "no pending session" occurs only once here, rather than repeatedly as per my first test, and the keep alive connection info continues after that point also, so the logging isn't behaving the same way after a service restart and resuming an existing session, than it does from a reboot of machine and new moonlight session:

> [2025-04-05 22:33:04.786]: Debug:  [--] 
> [2025-04-05 22:33:05.263]: Debug: /CN=NVIDIA GameStream Client -- verified
> [2025-04-05 22:33:05.264]: Debug: TUNNEL :: HTTPS
> [2025-04-05 22:33:05.264]: Debug: METHOD :: GET
> [2025-04-05 22:33:05.264]: Debug: DESTINATION :: /serverinfo
> [2025-04-05 22:33:05.264]: Debug: User-Agent -- okhttp/3.12.13
> [2025-04-05 22:33:05.264]: Debug: Accept-Encoding -- gzip
> [2025-04-05 22:33:05.264]: Debug: Connection -- Keep-Alive
> [2025-04-05 22:33:05.264]: Debug: Host -- xxx.xxx.xxx.xxx:47984
> [2025-04-05 22:33:05.264]: Debug:  [--] 
> [2025-04-05 22:33:05.264]: Debug: uuid -- f2482be3-3044-4c49-adee-356eb4749cfc
> [2025-04-05 22:33:05.264]: Debug: uniqueid -- 0123456789ABCDEF
> [2025-04-05 22:33:05.264]: Debug:  [--] 
> **[2025-04-05 22:33:06.648]: Debug: No pending session for incoming RTSP connection**

Here's the nvlddmkm error from event viewer that happens at the same time as the sunshine application error in windows event logs:
> The description for Event ID 153 from source nvlddmkm cannot be found. Either the component that raises this event is not installed on your local computer or the installation is corrupted. You can install or repair the component on the local computer.
> 
> If the event originated on another computer, the display information had to be saved with the event.
> 
> The following information was included with the event: 
> 
> \Device\000000d5
> Error occurred on GPUID: a00
> 
> The message resource is present but the message was not found in the message table

I briefly looked around for this and did find some similar errors, but not exactly the same.  I will look around further after finishing my next test.

I was also able to pull my old sunshine version out of a backup from my previous win10 install, it was version 0.23.1 from April 21 2024.  I will see if I can't pull the Nvidia driver version at the time out of backups also before I post all the info from testing.  Might be a day or two now, easter holidays, kids, etc.  Assuming I can, and after I've finished these tests, I will roll sunshine back to that version first and test again to see if that fixes it, and if not roll the nvidia driver version back after taking sunshine back up to the correct version.

Will continue on with this hoping to find the cause, as I simply must get this sorted.  Sunshine has been a godsend since gamestream went away and totally rock solid till now, and with win10 support ending in the near future important to get this sorted under 11, and looks like others are having the same issue on other OSs.  Will do everything I can to help iron out bugs and find conflicts with Nvidia drivers, which it looks like could be the issue here.

Would anyone watching or commenting on this thread mind posting their current nvidia card model and driver version please, along with sunshine version.  I think that could help in narrowing this down.  Thanks.

**Reply by BuckRogers25** [reply]
Found some closed issue threads on this earlier that are relevant, essentially the same issue.
https://github.com/LizardByte/Sunshine/issues/1031
https://github.com/LizardByte/Sunshine/issues/863

So is a known issue, and does look like driver + sunshine rather than sunshine just on its own.  

Disabling HAGS in display settings is a work around for 30-series users with decent CPUs, as HAGS isn't required for DLSS2 or 3 without frame generation.  May lead to a tiny drop in FPS and latency apparently, but haven't tested yet, and some users reporting even with HAGS disabled its still an issue.  It isn't a work around for 40-series users wanting to use frame generation however, as HAGS is required for framegen.

Again shall test more later.  This maybe needs adding to one of those issue threads instead of continuing here.

[a comment by an automated account is not shown]
